src/main/java/it/trade/api/StatelessTradeItApiClient.java

```diff
@@ -311,4 +311,57 @@ public void onSuccessResponse(Response<TradeItGetAllTransactionsHistoryResponse>
             }
         });
     }
+
+    public void previewCryptoOrder(
+            TradeItPreviewCryptoOrderRequest request,
+            final TradeItCallback<TradeItPreviewCryptoOrderResponse> callback
+    ) {
+        tradeItApi.previewCryptoOrder(request)
+                .enqueue(
+                        new PreviewTradeCallback<TradeItPreviewCryptoOrderResponse, TradeItPreviewCryptoOrderResponse>(
```

Comment:
Why does this take two types? Also what's different between `TradeItCallback` and `PreviewTradeCallback`? Is there a chance to eliminate one eventually or is it serving a useful purpose?

Reply:
`PreviewTradeCallback` deals with the trade it response to detect `REVIEW_ORDER` and return `onSuccessResponse`: https://github.com/tradingticket/JavaApi/blob/develop/src/main/java/it/trade/model/callback/PreviewTradeCallback.java#L22
It is used to simplify the code in the TradeItApiClient implementation

`TradeItCallback` is the callback the partner has to use when calling the api with `TradeItApiClient` or `StatelessApiClient`, to get the `onSuccess`/ `onError`.

The 2 types are:
- the first one is the trade it response type
- the second is the type we want to return in the TradeItCallback

Indeed here they are the same, but it could happen to be different for some callback.
Example 2 different types: https://github.com/tradingticket/JavaApi/blob/develop/src/main/java/it/trade/api/StatelessTradeItApiClient.java#L99
https://github.com/tradingticket/JavaApi/blob/develop/src/main/java/it/trade/api/StatelessTradeItApiClient.java#L267

The idea is to simplify the response in the tradeit callback for the partner.

Reply:
Cool - thanks for the explanation